recipes/boost/all/test_package/conanfile.py

```diff
@@ -31,9 +44,9 @@ def build(self):
                 cmake.definitions["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared
             cmake.definitions["WITH_PYTHON"] = not self.options["boost"].without_python
             if not self.options["boost"].without_python:
-                pyversion = tools.Version(self.options["boost"].python_version)
-                cmake.definitions["Python_ADDITIONAL_VERSIONS"] = "{}.{}".format(pyversion.major, pyversion.minor)
-                cmake.definitions["PYTHON_COMPONENT_SUFFIX"] = "{}{}".format(pyversion.major, pyversion.minor)
+                (v, _) = self._all_pythons()[0]
+                cmake.definitions["Python_ADDITIONAL_VERSIONS"] = "{}.{}".format(v.major, v.minor)
+                cmake.definitions["PYTHON_COMPONENT_SUFFIX"] = "{}{}".format(v.major, v.minor)
```

Comment:
This (somewhat arbitrary) selection of python lib to link against means *all* enabled pythons must have their equivalent pythonXY-devel package installed.

Otherwise no /usr/lib/libpythonXY.so and cmake configure fails with errors of the form:

```
CMake Error at CMakeLists.txt:113 (target_link_libraries):
  Target "hello_ext" links to:

    Boost::python39

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

```